src/helpers/color-helpers.ts

```diff
@@ -0,0 +1,19 @@
+type ColorMethod = 'hex' | 'rgb' | 'rgba' | 'hsl' | 'hsla' | 'named';
+
+export const isValidColor = (color: string, method?: ColorMethod): boolean => {
+  const patterns: Record<ColorMethod, RegExp> = {
+    hex: /^#(?:[\dA-Fa-f]{3}){1,2}$/,
+    rgb: /^rgb\((\d{1,3}%?,\s*){2}\d{1,3}%?\)$/,
+    rgba: /^rgba\((\d{1,3}%?,\s*){3}[\d.]+\)$/,
+    hsl: /^hsl\(\d{1,3}(?:,\s*\d{1,3}%){2}\)$/,
+    hsla: /^hsla\(\d{1,3}(?:,\s*\d{1,3}%){2},\s*[\d.]+\)$/,
+    named: /^[A-Za-z]+$/,
+  };
+
+  if (method) {
+    return patterns[method].test(color);
+  }
+
+  // Check all methods if no specific method is provided
+  return Object.values(patterns).some(regex => regex.test(color));
```

Comment:
Do we want to be more precise in the way we test the values, or shall we assume for now that users are giving correct values if they have managed to pass that regex?

Reply:
If we specify the regex then we obligatory test against that regex, otherwise (if we dont specify) we test all regexes and if one is true then it is, in fact, a color. No?

Reply:
Oh sorry forget I said that, I just did some tests and realised that even if the values are not in the expected range (like not between 0 and 255 for rgb) then it doesn't break anything, so there is no need to check the range of the values for each way colours can be put.